### ISSUE TITLE
enableRBDStats bug

### ISSUE THREAD

**Issue by BlaineEXE** [rook/rook#12627]
If a CephBlockPool's `enableRBDStats` field is changed from `true` to `false`, the pool will still remain in the `mgr/prometheus/rbd_stats_pools` list.

_Originally posted by @BlaineEXE in https://github.com/rook/rook/issues/11099#issuecomment-1658964154_

This may be a little complex for a first issue, but new technical contributors who are familiar with controller logic might find it a good place to start.

**Comment by BlaineEXE** [comment]
Still a bug, but very minor

**Comment by OdedViner** [comment]
Tested manually:
Procedure:
1. Start minikube wiht kvm2:
```
$ minikube start --disk-size=10g --extra-disks=1 --driver kvm2
```

2.Create resources and operator
```
$ kubectl create -f crds.yaml -f common.yaml -f operator.yaml
$ kubectl get pods -n rook-ceph
NAME                                 READY   STATUS    RESTARTS   AGE
rook-ceph-operator-cf4f7dfd4-vzwbt   1/1     Running   0          89s
```

3.Create cluster-test:
```
$ cd deploy/examples/
$ kubectl create -f cluster-test.yaml 
```

4.Create tool pod and check ceph status:
```
$ cd deploy/examples/
kubectl create -f toolbox.yaml
kubectl -n rook-ceph exec -it deploy/rook-ceph-tools -- bash
bash-5.1$ ceph health
HEALTH_OK
```

5.Create pool-test with [enableRBDStats: true]
```
kubectl create -f pool-test.yaml
```
```
apiVersion: ceph.rook.io/v1
kind: CephBlockPool
metadata:
  name: replicapool
  namespace: rook-ceph # namespace:cluster
spec:
  enableRBDStats: true
  failureDomain: osd
  replicated:
    size: 1
```

6.Check `mgr/prometheus/rbd_stats_pools` list
```
$ kubectl -n rook-ceph exec -it deploy/rook-ceph-tools -- bash
bash-5.1$  ceph config get mgr mgr/prometheus/rbd_stats_pools
replicapool
```

7.Change enableRBDStats to false
```
$ kubectl edit CephBlockPool replicapool -n rook-ceph 
 enableRBDStats: false
```

 8.Wait 20 min
 
 9.Check `mgr/prometheus/rbd_stats_pools` [the `replicapool`  did not remove from `mgr/prometheus/rbd_stats_pools` list]
```
bash-5.1$   ceph config get mgr mgr/prometheus/rbd_stats_pools
replicapool
```

**Comment by BlaineEXE** [comment]
Thanks for validating the bug is present! I've assigned the bug to you so that others are aware you're working on it.

**Comment by BlaineEXE** [comment]
Closed by #14897
